service/src/test/java/bio/terra/pipelines/common/utils/PipelineRunFilterSpecificationTest.java

```diff
@@ -0,0 +1,253 @@
+package bio.terra.pipelines.common.utils;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+import bio.terra.pipelines.db.entities.PipelineRun;
+import bio.terra.pipelines.service.exception.InvalidFilterException;
+import jakarta.persistence.criteria.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+class PipelineRunFilterSpecificationTest {
```

Comment:
might be nice to have some tests using the actual db where we add different pipeline runs to ensure the filters are working as expected (like the test you have testing pipeline filtered count